tests/Invoke-DbaQuery.Tests.ps1

```diff
@@ -1,14 +1,18 @@
-$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
-Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
-. "$PSScriptRoot\constants.ps1"
+BeforeAll {
+    $CommandName = (Get-Item $PSCommandPath).Name.Replace(".Tests.ps1", "")
+    Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
+    $PesterV5Throw = ''
+    if ((Get-Command Invoke-Pester -ErrorAction SilentlyContinue).Version -ge '5.0.0') {
+        $PesterV5Throw = '*'
+    }
+}
 
 Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
     Context "Validate parameters" {
-        [object[]]$params = (Get-Command $CommandName).Parameters.Keys | Where-Object { $_ -notin ('whatif', 'confirm') }
-        [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Database', 'Query', 'QueryTimeout', 'File', 'SqlObject', 'As', 'SqlParameters', 'AppendServerInstance', 'MessagesToOutput', 'InputObject', 'ReadOnly', 'EnableException', 'CommandType'
-        $knownParameters += [System.Management.Automation.PSCmdlet]::CommonParameters
         It "Should only contain our specific parameters" {
-            (@(Compare-Object -ReferenceObject ($knownParameters | Where-Object { $_ }) -DifferenceObject $params).Count ) | Should Be 0
+            [array]$params = ([Management.Automation.CommandMetaData]$ExecutionContext.SessionState.InvokeCommand.GetCommand($CommandName, 'Function')).Parameters.Keys
```

Comment:
You could probably move this to the BeforeAll could you not? (I see the test failed on it not seeing anything in the variable

Reply:
"global" scope. Lemme reconstruct the before-cleanup logic :D